assets/src/blocks/Articles/Articles.js

```diff
@@ -195,64 +203,70 @@ export class Articles extends Component {
             help={__('Open button link in new tab', 'p4ge')}
             value={this.props.button_link_new_tab}
             checked={this.props.button_link_new_tab}
-            onChange={(e) => this.props.onButtonLinkTabChange(e)}
+            onChange={ value => {
+              setAttributes( { button_link_new_tab: value } );
+            } }
           />
         </div>
 
         {
-          this.props.posts !== 'undefined' && this.props.posts.length === 0
+          this.props.attributes.posts !== 'undefined' && this.props.attributes.posts.length === 0
             ?
             <Fragment>
               <div>
                 <TextControl
                   label={__('Articles count', 'p4ge')}
                   help={__('Number of articles', 'p4ge')}
                   type="number"
-                  value={this.props.article_count}
-                  onChange={this.props.onCountChange}
+                  value={this.props.attributes.article_count}
+                  onChange={ value => {
+                    setAttributes( { article_count: value } );
+                  } }
                 />
               </div>
               <div>
                 <FormTokenField
-                  value={this.state.tagTokens}
                   suggestions={tagSuggestions}
                   label={__('Select Tags', 'p4ge')}
-                  onChange={tokens => this.onSelectedTagsChange(tokens)}
+                  value={this.state.tagTokens}
+                  onChange={this.onSelectedTagsChange.bind(this)}
```

Comment:
Please move these bindings to the constructor (already discussed via chat). We can tweak Babel to allow arrow bindings later.